src/app/modules/photo-uploader/components/uploader-content-card/uploader-content-card.component.ts

```diff
@@ -0,0 +1,129 @@
+  styleUrls: ['./uploader-content-card.component.scss']
+})
+export class UploaderContentCardComponent implements OnInit, OnDestroy {
+
+  @Input() fileItem: FileItem;
+  @Input() photoUploaderModel: PhotoUploaderModel;
+
+  public photoUploaderModelSubscription: any;
+  public photoUploaderModelChangeEditSubscription: any;
+
+  public filePreviewPath: SafeUrl;
+  uploadPhotoForm: FormGroup;
+  isEditMode = false;
+
+  constructor(
+    private sanitizer: DomSanitizer,
+    private formBuilder: FormBuilder,
+    private localStorageService: UploadPhotoLocalStorageService,
+    private photoEventService: PhotoEventService,
+    private dialog: MatDialog) { }
+
+  ngOnInit() {
+    this.prepareFilePreview();
+    this.createUploadPhotoForm();
+    this.propagateCurrentChosedPhoto();
+    this.subscireEvents();
+  }
+
+  ngOnDestroy() {
+    this.photoUploaderModelSubscription.unsubscribe();
+    this.photoUploaderModelChangeEditSubscription.unsubscribe();
+  }
+
+  refreshEditMode() {
+    const currentPhoto = this.localStorageService.getCurrentChosedPhoto();
+
+    if (this.photoUploaderModel.index !== currentPhoto.index) {
+      this.photoUploaderModel.isEditMode = false;
+    } else {
+      this.photoUploaderModel.isEditMode = true;
+    }
+  }
+
+  prepareFilePreview() {
+    this.filePreviewPath = this.getSafeUrl(this.fileItem._file);
+  }
+
+  createUploadPhotoForm() {
+    this.uploadPhotoForm = this.formBuilder.group({
+      photoTitle: [this.fileItem._file.name],
+      photoSubtitle: [''],
+      photoDescription: ['']
+    });
+  }
+
+  choosePhoto() {
+    const currentChosedPhotoWithEditMode = this.localStorageService.getCurrentChosedPhoto();
+    currentChosedPhotoWithEditMode.isEditMode = false;
+    this.localStorageService.updatePhoto(currentChosedPhotoWithEditMode);
+
+    this.photoUploaderModel.isEditMode = true;
+    this.localStorageService.updatePhoto(this.photoUploaderModel); // update phote after chosed
+
+    this.propagateCurrentChosedPhoto();
+  }
+
+  changePhotoTitle(event) {
+    const changedPhotoTitle = event.currentTarget.value;
+    this.photoUploaderModel.photoTitle = changedPhotoTitle;
+    this.syncPhotoChanges();
+  }
+
+  changePhotoDescription(event) {
+    this.photoUploaderModel.photoDescription = event.currentTarget.value;
+    this.syncPhotoChanges();
+  }
+
+  openDeleteConfirmationDialog() {
+    const dialogRef = this.dialog.open(DeleteConfirmationDialogComponent, {
+      data: {photoTitle: this.photoUploaderModel.photoTitle}
+    });
+    dialogRef.afterClosed().subscribe(isConfirmPhotoRemove => {
+      if (isConfirmPhotoRemove) {
+        this.photoEventService.emitphotoUploaderRemoveChosenPhoto(this.fileItem);
+      }
+    });
+  }
+
+  private syncPhotoChanges() {
+    this.localStorageService.updatePhoto(this.photoUploaderModel);
+    this.propagateCurrentChosedPhoto();
+  }
+
+  private updateCurrentPhoto(photo: PhotoUploaderModel) {
```

Comment:
What about photoTags?

Reply:
ts will be extended in the features with sc path for fileUpload. now it's only skeleton :)